Port to LLVM 11.0.0 + basic refactoring via Resharper

leskovde opened the pull request:
Code cleanup, setup summary and changes to the newest LLVM / libTooling framework.

Commit:
Ported to LLVM 10.

Commit:
Code cleanup, added one test case.

Commit:
LLVM 11.0.0: Fixed additional libraries to support LLVM 11.0.0.

Commit:
Added setup manual and dependency info.

dwat3r commented on test/dot2png.bat:
```diff
@@ -1 +1 @@
-for %%v in (*.dot) do dot -Tpng "%%v" -o "%%~nv.png"
\ No newline at end of file
+for %%v in (*.dot) do "C:\Program Files\Graphviz 2.44.1\bin\dot.exe" -Tpng "%%v" -o "%%~nv.png"
```
Please make this a relative path, like 'dot.exe', for portability.

dwat3r reviewed (changes requested):
Hey, thanks for updating my project!

I have a couple of suggestions:
Please use 2 spaces when formatting.
, 4 spaces makes the code too wide.

Please don't delete png files, as they are important showcase what this tool can do.

Could you tell me why do you need this PR to be merged? I'm asking out of curiosity. Updating is nice really.

Commit:
Changed code spacing to 2 spaces.

Commit:
PNG files reset, removed absolute paths from dot2png.bat.

Commit:
Removed relative dependencies created by me and simplified the setup manual.

Commit:
Changed one additional absolute path.

leskovde commented:
Hello,
I've reverted all the PNG files I've modified or removed on accident as well as the formatting of the code. 

I'm working on a bachelor thesis. During my research, I was looking for Clang-based slicers and yours is one of the few I've found. I tried to run the code on the latest release of LLVM, however, due to frequent changes to Clang and LibTooling, it just wouldn't compile. 
I've decided that it would be useful for others to update the project to the latest release, rather than download an outdated version of LLVM (which, as I've found out, is not always possible on some systems).

Thanks for making this project available. I'll make sure to reference it in my work :)

Cheers!

dwat3r commented:
Thank you, this work was my masters thesis, so I'm happy that you've found it useful for yours! :)
Thank you again putting effort into it.
The thing is that it has limited slicing capabilities, so a lot of improvements could be made, but I was lazy to work on it after finishing MSc.

One of the challenges are to make it work with classes, and to slice the whole compilation unit, across functions, excluding standard libraries.

The PR now looks now great, so I'll merge it.
